src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -144,13 +151,23 @@ await functionMetadata
                 await FileUtility.WriteAsync(dataFilePath, functionMetadata.TestData);
             }
 
+            var updatedFunctionMetadata = (await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(_languageWorkerOptions.CurrentValue.WorkerConfigs, configChanged))
+                .FirstOrDefault(metadata => Utility.FunctionNamesMatch(metadata.Name, name));
+
+            bool getFunctionMetadataSuccessful = false;
+            FunctionMetadataResponse functionMetadataResult = null;
+            if (updatedFunctionMetadata != null)
+            {
+                string routePrefix = await GetRoutePrefix(hostOptions.RootScriptPath);
```

Comment:
I think the following 3 lines are a copy of the code [here](https://github.com/Azure/azure-functions-host/blob/ba3e9ef990a9a5ebacaf47bbc3a71ed5f661e8ea/src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs#L182) in this same file. Can you extract into a common method `GetFunctionMetadataResponseAsync(hostOptions, functionMetadata, request)`?